### ISSUE TITLE
WSProto: cannot close 403 during handshake

### ISSUE THREAD

**Issue by Andrew-Chen-Wang** [Kludex/uvicorn#811]
https://github.com/encode/uvicorn/blob/d755fe850d2bf740c5dd091e091e154457634c5f/uvicorn/protocols/websockets/wsproto_impl.py#L249-L255

If I have an application like this:

```python
async def app(scope, receive, send):
    event = await receive() 
    # Something rejecting happens
    # For reproducing purposes, know that I did not send anything else beforehand, not even websocket.accept. The only sent thing is the below
    await send({"type": "websocket.close"})
```
WSProto will raise an error:
```python
wsproto.utilities.LocalProtocolError: Event Response(status_code=403, headers=<Headers([])>, http_version=b'1.1', reason=b'') cannot be sent during the handshake
```

This is in reference to "how do I reject a connection?" I've also tested this with `websockets`package, all including uvicorn on their latest versions, but this only applies to wsproto as in it works fine with the websockets package.

Note: I tried just doing a `return` instead of `websocket.close` but that results in (L219 in protocols/websockets/wsproto_impl:

```
ASGI callable returned without completing handshake.
```

since the handshake was never set to `True`.

**Comment by Andrew-Chen-Wang** [comment]
If it's confusing, it really just means your code is sending a status 403 but WSProto will raise an error crashing uvicorn.

**Comment by Andrew-Chen-Wang** [comment]
Is there something confusing here...? If you try to decline the connection (i.e. don't accept) by simply closing the connection, WSProto raises an error. The only workaround for this would be to accept the connection then immediately close it (e.g. non authenticated users), but that would double the data since we accept then close rather than immediately close.

**Comment by adamhooper** [comment]
This looks like #244 ... fixed by #704.

**Comment by florimondmanca** [comment]
Yup, this should be fixed by #704.

**Comment by Andrew-Chen-Wang** [comment]
Thanks for fixing this!
